.github/workflows/e2e.yaml

```diff
@@ -47,7 +47,33 @@ jobs:
       - name: Harden Runner
         uses: step-security/harden-runner@5c7944e73c4c2a096b17a9cb74d65b6c2bbafbde # v2.9.1
         with:
-          egress-policy: audit
+          disable-telemetry: true
+          egress-policy: block
+          allowed-endpoints: >
+            *.azmk8s.io:443
+            *.azurecr.io:443
+            *.data.mcr.microsoft.com:443
```

Comment:
Maybe call these out in a comment too, since they are our only regex ones?

The PR description is very clarifying for all the odd cases, but that won't directly stick around with the code, and will be harder to fine in the future.